registry/storage/cache/memory/memory.go

```diff
@@ -57,6 +65,10 @@ func (imbdcp *inMemoryBlobDescriptorCacheProvider) RepositoryScoped(repo string)
 }
 
 func (imbdcp *inMemoryBlobDescriptorCacheProvider) Stat(ctx context.Context, dgst digest.Digest) (distribution.Descriptor, error) {
+	span, _ := tracing.StartSpan(ctx, fmt.Sprintf("%s:%s", TraceSpanCacheType, "Stat"),
```

Comment:
Concatenate string constants with the `+` operator, not `fmt.Sprintf`, so the compiler can evaluate the concatenation at build time. And do not discard the span context. It should always overwrite the ctx argument so it is always passed into called functions.
```suggestion
	span, ctx := tracing.StartSpan(ctx, TraceSpanCacheType +":Stat",
```